src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -188,104 +188,108 @@ class DefaultSource
-      relation
-    } else {
-      val relation = resourceType match {
-        case "events" =>
-          new EventsRelation(config)(sqlContext)
-        case "timeseries" =>
-          new TimeSeriesRelation(config)(sqlContext)
-        case "assets" =>
-          new AssetsRelation(config)(sqlContext)
-        case "files" =>
-          new FilesRelation(config)(sqlContext)
-        case "sequences" =>
-          new SequencesRelation(config)(sqlContext)
-        case "labels" =>
-          new LabelsRelation(config)(sqlContext)
-        case "sequencerows" =>
-          createSequenceRows(parameters, config, sqlContext)
-        case "relationships" =>
-          new RelationshipsRelation(config)(sqlContext)
-        case "datasets" =>
-          new DataSetsRelation(config)(sqlContext)
-        case FlexibleDataModelRelationFactory.ResourceType =>
-          createFlexibleDataModelRelation(parameters, config, sqlContext)
-        case _ => sys.error(s"Resource type $resourceType does not support save()")
-      }
-      val batchSizeDefault = relation match {
-        case _: SequenceRowsRelation => Constants.DefaultSequenceRowsBatchSize
-        case _ => Constants.DefaultBatchSize
-      }
-      val batchSize = config.batchSize.getOrElse(batchSizeDefault)
-      val originalNumberOfPartitions = data.rdd.getNumPartitions
-      val idealNumberOfPartitions = config.sparkPartitions
-
-      // If we have very many partitions, it's quite likely that they are significantly uneven.
-      // And we will have to limit parallelism on each partition to low number, so the operation could
-      // take unnecessarily long time. Rather than risking this, we'll just repartition data in such case.
-      // If the number of partitions is reasonable, we avoid the data shuffling
-      val dataRepartitioned =
-        if (originalNumberOfPartitions > 50 && originalNumberOfPartitions > idealNumberOfPartitions) {
-          data.repartition(idealNumberOfPartitions)
-        } else {
-          data
-        }
-
-      dataRepartitioned.foreachPartition((rows: Iterator[Row]) => {
-        import CdpConnector.ioRuntime
 
-        val maxParallelism = config.parallelismPerPartition
-        val batches = Stream.fromIterator[IO](rows, chunkSize = batchSize).chunks
+    val relation: CdfRelation = resourceType match {
+      case AssetHierarchyBuilder.name =>
+        new AssetHierarchyBuilder(config)(sqlContext)
+      case NumericDataPointsRelation.name =>
+        new NumericDataPointsRelationV1(config)(sqlContext)
+      case StringDataPointsRelation.name =>
+        new StringDataPointsRelationV1(config)(sqlContext)
+      case EventsRelation.name =>
+        new EventsRelation(config)(sqlContext)
+      case TimeSeriesRelation.name =>
+        new TimeSeriesRelation(config)(sqlContext)
+      case AssetsRelation.name =>
+        new AssetsRelation(config)(sqlContext)
+      case FilesRelation.name =>
+        new FilesRelation(config)(sqlContext)
```

Comment:
I guess that we handle it here and don't bail with "doesn't support save()" might be ok in presence of another check later on, but looks a bit odd

Reply:
nevermind, that's files metadata, not contents